app/helpers/reactjs_helper.rb

```diff
@@ -65,6 +65,14 @@ def css_tags_for(requested_plugins)
     end
   end
 
+  def locale_js_tags
+    domains = ::FastGettext.translation_repositories.keys
+    plugins = ::Foreman::Plugin.all.map { |p| p.engine.class.to_s.underscore.split('/', 2).first }
```

Comment:
Do you mean to extract the plugin name? I tihnk you can use `Plugin#id`

Reply:
A lot of this has been iterated on in https://github.com/theforeman/foreman/pull/9692 so I'd suggest to review that first. I think it's very close to being done and then for this PR to be rebased.